### PR TITLE
refactoring IEmailSender and ISmsSender out of AdminController

### PR DESCRIPTION
**Pull request by mgmccarthy**
- removed IEmailSender and ISmsSender from AdminController and replaced with Mediator
- added messsage/handlers for each email/sms sent from controller that is now being sent by Mediator
- updated unit tests for AdminControllerTests
- **TODO**: add unit tests for newly added handlers (this will be a separate PR)

### CODE REVIEW

**Comment by tonysurma** [comment]
my review looks good, @MisterJames can you give this a quick once over?

**Comment by mgmccarthy** [comment]
@tonysurma @MisterJames I'd work on merging #723 (Final renaming of activities to events) before merging this. I can easily rebase and fix and naming conflicts, then push. I'll have a separate PR for the rest of the unit tests associated to work done in this PR.

**Comment by tonysurma** [comment]
I merged that PR @mgmccarthy

**Comment by mgmccarthy** [comment]
@tonysurma, looks like we're good on the local rebase off of the renaming's. My guess is the renaming's did not touch any of this code in the PR. Should be good to go whenever a code review can be done.

File: AllReadyApp/Web-App/AllReady.UnitTest/Controllers/AdminControllerTests.cs [hunk 1]
```diff
@@ -107,7 +108,8 @@ public void RegisterHasAllowAnonymousAttribute()
             var userManager = CreateUserManagerMock();
             userManager.Setup(x => x.CreateAsync(It.IsAny<ApplicationUser>(), It.IsAny<string>())).Returns(() => Task.FromResult(IdentityResult.Success));
 
-            var sut = new AdminController(userManager.Object, null, Mock.Of<IEmailSender>(), null, null, generalSettings.Object);
+            //var sut = new AdminController(userManager.Object, null, Mock.Of<IEmailSender>(), null, null, generalSettings.Object);
```
Comment (MisterJames) [inline AllReadyApp/Web-App/AllReady.UnitTest/Controllers/AdminControllerTests.cs]: We can probably just nuke this (instead of commenting it out).

**Comment by MisterJames** [comment]
Okay, good PR. In reviewing this I'm thinking I'm going to merge, but we're continuing to work against two pieces of tech debt here: Admin-based account confirmation and messaging propagating it's way through the system (rather than being abstracted out).

I'm going to think on a couple of things:
 1. Unifying the signup/confirmation process (we can always elevate to admin with claims through some other process)
 1. Getting a templating process in place for emails/SMS such that templates and related language are defined external to the source code (but preferably under source control)
 1. Ensuring users can manage notification preferences (SMS, email, digest, DND)
 1. Pushing the code base to use a unified messaging service/command in such a way that the caller doesn't need to know how the user prefers to be contacted.

File: AllReadyApp/Web-App/AllReady.UnitTest/Controllers/AdminControllerTests.cs (hunk 1 above)
Comment (mgmccarthy) [inline AllReadyApp/Web-App/AllReady.UnitTest/Controllers/AdminControllerTests.cs]: i can take care of this in another PR. Sorry, I forgot to uncomment it.

**Comment by mgmccarthy** [comment]
:+1: for all of those, especially the unified message service/command that figures out how the user want to be notified instead of the caller need to know which type of notification to use.

That can be pulled from the users managing their notification process in another page.

If you need help or want to delegate, let me know.

Thanks

**Pull request merged**